dspace-server-webapp/src/test/java/org/dspace/app/rest/CommunityAdminGroupRestControllerIT.java

```diff
@@ -0,0 +1,351 @@
+    }
+
+
+    @Test
+    public void getCommunityAdminGroupUnAuthorizedTest() throws Exception {
+        communityService.createAdministrators(context, parentCommunity);
+
+        getClient().perform(get("/api/core/communities/" + parentCommunity.getID() + "/adminGroup"))
+                   .andExpect(status().isUnauthorized());
+    }
+
+    @Test
+    public void getCommunityAdminGroupForbiddenTest() throws Exception {
+        communityService.createAdministrators(context, parentCommunity);
+
+        String token = getAuthToken(eperson.getEmail(), password);
+        getClient(token).perform(get("/api/core/communities/" + parentCommunity.getID() + "/adminGroup"))
+                        .andExpect(status().isForbidden());
+    }
+
+    @Test
+    public void getCommunityAdminGroupNoContentTest() throws Exception {
+
+        String token = getAuthToken(admin.getEmail(), password);
+        getClient(token).perform(get("/api/core/communities/" + parentCommunity.getID() + "/adminGroup"))
+                        .andExpect(status().isNoContent());
+    }
+
+    @Test
+    public void getCommunityAdminGroupWrongCommunityUuidResourceNotFoundTest() throws Exception {
+
+        String token = getAuthToken(admin.getEmail(), password);
+        getClient(token).perform(get("/api/core/communities/" + UUID.randomUUID() + "/adminGroup"))
+                        .andExpect(status().isNotFound());
+    }
+
+    @Test
+    public void postCommunityAdminGroupCreateAdminGroupSuccess() throws Exception {
+
+        ObjectMapper mapper = new ObjectMapper();
+        GroupRest groupRest = new GroupRest();
+        MetadataRest metadataRest = new MetadataRest();
+        metadataRest.put("dc.description", new MetadataValueRest("testingDescription"));
+        metadataRest.put("dc.subject", new MetadataValueRest("testSubject"));
+
+        groupRest.setMetadata(metadataRest);
+
+        AtomicReference<UUID> idRef = new AtomicReference<>();
+
+        String token = getAuthToken(admin.getEmail(), password);
+        getClient(token).perform(post("/api/core/communities/" + parentCommunity.getID() + "/adminGroup")
+                                     .content(mapper.writeValueAsBytes(groupRest))
+                                     .contentType(contentType))
+                        .andExpect(status().isCreated())
+                        .andDo(result -> idRef
+                            .set(UUID.fromString(read(result.getResponse().getContentAsString(), "$.id")))
+                        );
+        Group adminGroup = groupService.find(context, idRef.get());
+        getClient(token).perform(get("/api/eperson/groups/" + adminGroup.getID()))
+                        .andExpect(status().isOk())
+                        .andExpect(
+                            jsonPath("$", GroupMatcher.matchGroupEntry(adminGroup.getID(), adminGroup.getName())));
```

Comment:
Again, the POSTed Group in this method needs to be deleted at the end of the method.